superset/mcp_service/app.py

```diff
@@ -0,0 +1,100 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+FastMCP app instance and initialization for Superset MCP service.
+This file provides the global FastMCP instance (mcp) and a function to initialize
+the server. All tool modules should import mcp from here and use @mcp.tool decorators.
+"""
+
+import logging
+
+from fastmcp import FastMCP
+
+logger = logging.getLogger(__name__)
+
+# Create MCP instance without auth for scaffold
+mcp = FastMCP(
```

Comment:
It would be great if we could have a configurable factory to create the FastMCP instance, this way users can freely configure it the way they want/need, add custom Auth, Middleware etc. Copying the Flask pattern.
https://gofastmcp.com/servers/server#creating-a-server